crates/kubelet/src/backoff/mod.rs

```diff
@@ -0,0 +1,104 @@
+//! Provides backoff timing control for Kubernetes pod states
+//! such as ImagePullBackoff and CrashLoopBackoff.
+use std::time::Duration;
+
+/// Determines how long to back off before performing a retry.
+#[async_trait::async_trait]
+pub trait BackoffStrategy: Send {
+    /// Resets the strategy after a success.
+    fn reset(&mut self);
+    /// Gets how long to wait before retrying.
+    fn next_duration(&mut self) -> Duration;
+    /// Waits the prescribed amount of time (as per `next_duration`).
+    async fn wait(&mut self) {
+        tokio::time::delay_for(self.next_duration()).await
+    }
+}
+
+/// A `BackoffStrategy` in which the durations increase exponentially
+/// until hitting a cap.
+pub struct ExponentialBackoffStrategy {
+    base_duration: Duration,
+    cap: Duration,
+    last_duration: Duration,
+}
+
+impl ExponentialBackoffStrategy {
+    /// Gets a backoff strategy that adheres to the Kubernetes defaults.
+    pub fn kubernetes_default() -> Self {
```

Comment:
Should we just make this a `Default` implementation instead or do we intend for this to be used in other non-krustlet contexts?

Reply:
I kind of wanted to explicitly make it clear that this was the default _for the context of Kubernetes_.  But you're right.  There _is_ no context other than Kubernetes, so this can just be `default`.